packages/react-reconciler/src/ReactAbortController.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {enableCache} from 'shared/ReactFeatureFlags';
+
+const PossiblyWeakMap = typeof WeakMap === 'function' ? WeakMap : Map;
+
+const signals: Map<
+  AbortControllerShim,
+  AbortSignalShim,
+> = (new PossiblyWeakMap(): any);
+
+const abortedFlags: Map<
+  AbortSignalShim,
+  boolean,
+> = (new PossiblyWeakMap(): any);
+
+type Listener = {
+  listener: any => void,
+  next: Listener | null,
+};
+
+const listenersMap: Map<
+  AbortSignalShim,
+  Map<string, Listener>,
+> = (new PossiblyWeakMap(): any);
+
+function getListeners(target): Map<string, Listener> {
+  const listeners = listenersMap.get(target);
+  if (listeners == null) {
+    // eslint-disable-next-line react-internal/prod-error-codes
+    throw new Error(
+      'Listeners should already be registered. This is a bug in React.',
+    );
+  }
+  return listeners;
+}
+
+class AbortSignalShim {
```

Comment:
@acdlite is this what you had in mind? I need to include a basic event emitter for Node